backend/services/useradm/store/mongo/bson.go

```diff
@@ -0,0 +1,24 @@
+package mongo
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/mendersoftware/mender-server/services/useradm/model"
+	"go.mongodb.org/mongo-driver/bson/bsoncodec"
+	"go.mongodb.org/mongo-driver/bson/bsonrw"
+)
+
+var tEmail = reflect.TypeOf(model.Email(""))
```

Comment:
do you think there is a solution that does not involve reflection?

Reply:
I couldn't think of any that would not pull `go.mongodb.org/mongo-driver` as a dependency outside the `mongo` package.

Reply:
perhaps making `pkg/store` or similar is a good idea?

Reply:
It's extending `github.com/mendersoftware/mender-server/pkg/mongo/codec` with an additional encoding of `model.Email` type which is local to this service. I'm not sure what else I should to do?